### ISSUE TITLE
Extensions to the Paper

### ISSUE THREAD

**Issue by williamleach** [GeWu-Lab/OGM-GE_CVPR2022#16]
Hello,

Thank you providing this useful repo, I enjoyed reading your paper and the work is really interesting. I had a few questions about extensions to this work that I could not find the answers to in the paper, if you have any insight that would be greatly appreciated:

1. How would this method extend to more than 2 modalities? The discrepancy ratio equations are set up for a binary situation, could the ratio be calculated for 3 or more modalities?

2. Would this method work for multi-task output? Say the network outputs a binary cat/dog output, and a separate black/brown/white fur output. How would on calculate the scores then in eqs. 8 of the paper, as each input would have a multi-label output.

3. I am doing training with individual learning rate scaling of the params in each modality's encoder (initial lr of visual encoder is 0.1, initial lr of audio encoder is 0.01 for example), how would the calculation of the coefficients in eq. 10 of the paper be changed? Or should this not factor into the modulation?

Thanks again for any information you may have.

**Comment by echo0409** [comment]
Hi, thanks for your questions.

1. More than two modalities are a valuable extension,  while we have not conducted related experiments yet. I consider that one simple way is that calculated the one-to-others discrepancy ratios pair-wise, then average the results.

2. Also, we have not attempted the multi-task scenarios yet. The multi-task scenarios are more complex since the balance among losses should also be taken into consideration.

3. The learning rate and batch size are two factors affecting SGD noise. Hence, when the value of the learning rate of two modalities are different, the gradient modulation degree is ( eta*k, eta is the learning rate, equ 11).
